disnake/guild.py

```diff
@@ -2224,14 +2232,29 @@ async def create_scheduled_event(
             The request failed.
         TypeError
             The ``image`` asset is a lottie sticker (see :func:`Sticker.read`),
-            or one of ``entity_type``, ``privacy_level``, or ``entity_metadata``
-            is not of the correct type.
+            one of ``entity_type``, ``privacy_level``, or ``entity_metadata``
+            is not of the correct type, or the ``entity_type`` was not provided and
+            could not be assumed from the ``channel``.
 
         Returns
         -------
         :class:`GuildScheduledEvent`
             The newly created guild scheduled event.
         """
+
+        if channel is None:
+            entity_type = GuildScheduledEventEntityType.external
+            channel = MISSING
+        elif hasattr(channel, "type") and isinstance(channel.type, ChannelType):  # type: ignore
+            if channel.type == ChannelType.voice:  # type: ignore
+                entity_type = GuildScheduledEventEntityType.voice
+            elif channel.type == ChannelType.stage_voice:  # type: ignore
+                entity_type = GuildScheduledEventEntityType.stage_instance
```

Comment:
```suggestion
        elif isinstance(channel_type := getattr(channel, "type", None), ChannelType):
            if channel_type == ChannelType.voice:
                entity_type = GuildScheduledEventEntityType.voice
            elif channel_type == ChannelType.stage_voice:
                entity_type = GuildScheduledEventEntityType.stage_instance
```
To avoid the `# type: ignore`s :p